numpy/typing/_dtype_like.py

```diff
@@ -28,17 +56,7 @@
     List[Any],
     # {'names': ..., 'formats': ..., 'offsets': ..., 'titles': ...,
     #  'itemsize': ...}
-    # TODO: use TypedDict when/if it's officially supported
-    Dict[
-        str,
-        Union[
-            Sequence[str],  # names
-            Sequence[_DtypeLikeNested],  # formats
-            Sequence[int],  # offsets
-            Sequence[Union[bytes, str, None]],  # titles
-            int,  # itemsize
-        ],
-    ],
+    _DtypeDict,
     # {'field1': ..., 'field2': ..., ...}
     Dict[str, Tuple[_DtypeLikeNested, int]],
```

Comment:
maybe not related to this PR, but looks like this type of usage is discouraged in numpy ( https://numpy.org/devdocs/reference/arrays.dtypes.html ). Would static types be a good place to error on this usage if it is not encouraged ?

Reply:
> Would static types be a good place to error on this usage if it is not encouraged ?

I would argue that it is, though I feel such a change will require some discussion beforehand.

Reply:
okay, i will open an issue.

Reply:
added an issue: https://github.com/numpy/numpy/issues/16891

Reply:
Based on the discussion in #16891, we can remove this line and move https://github.com/numpy/numpy/pull/16622/files#diff-ccb6649c3fc407a2b5413ef6f80f1dfcR28 to numpy/tests/typing/fail/dtype.py ?

Reply:
And done as of https://github.com/numpy/numpy/pull/16622/commits/1c701a7cbc51c04d30d35767adc78863fabe4542.